compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2468,17 +2468,36 @@ private void createSimpleVarDefStmt(BLangSimpleVariable simpleVariable, BLangBlo
 
     @Override
     public void visit(BLangAssignment assignNode) {
-        boolean fieldAccessLVExpr = assignNode.varRef.getKind() == NodeKind.FIELD_BASED_ACCESS_EXPR;
+        boolean isOptionalFieldAssignment = isOptionalBasicTypeFieldAssignment(assignNode);
```

Comment:
The name doesn't convey the entire check, right? Also see https://stackoverflow.com/questions/5322648/for-a-boolean-field-what-is-the-naming-convention-for-its-getter-setter.

Reply:
Renamed to `shouldWidenExpressionTypeWithNil` and factored out the common part to `isOptionalFieldAssignment` (reused in `SemanticAnalyzer`)